src/CloneExtensions.UnitTests/ComplexTypeTests.cs

```diff
@@ -133,5 +145,15 @@ class CircularReference2
         {
             public CircularReference1 Other { get;set; }
         }
+
+        class BaseClass
+        {
+            public int Property { get; set; }
+        }
+
+        class DerivedClass1 : BaseClass
+        {
+            public new int Property { get; set; }
```

Comment:
Couple other interesting cases:
- `abstract` property with an implementation in derived class
- `virtual` property with override in derived class

Would you mind adding tests for that too?

Reply:
I never thought of the virtual and abstract properties ...

After some testing I found that virtual and abstract properties, unlike "new" properties, do not "hide" what they override ... they override it.  That is, if you set/get a virtual/abstract property on the base class or it on the sub class they manipulate the same values unlike "new"

I updated the testing class as requested with asserts to demonstrate what I indicated above. 

The code, as it is now, works for properties ... but both the virtual/abstract properties in the base class AND the sub class are fetched and set leading to redundant work.  Really, what we need is a way to find only the upper most property for virtual/abstract properties and only set that.  I am not sure how to do that at the moment.